idea/src/org/jetbrains/kotlin/idea/inspections/RedundantExplicitTypeInspection.kt

```diff
@@ -20,7 +21,7 @@ class RedundantExplicitTypeInspection : AbstractKotlinInspection() {
             if (!property.isLocal) return
             val typeReference = property.typeReference ?: return
             val initializer = property.initializer ?: return
-
+            if ((typeReference.typeElement as? KtUserType)?.referenceExpression?.mainReference?.resolve() is KtTypeAlias) return
```

Comment:
I'm almost sure this can be done easier, without additional reference resolve. Probably `type` should be `WrappedType` for type alias (not sure exactly).